src/wallet/wallet.cpp

```diff
@@ -2466,12 +2458,8 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             if (coin.m_input_bytes <= 0) {
                 return false; // Not solvable, can't estimate size for fee
             }
-            coin.effective_value = coin.txout.nValue - coin_selection_params.m_effective_feerate.GetFee(coin.m_input_bytes);
-            if (coin_selection_params.use_bnb) {
-                value_to_select -= coin.effective_value;
-            } else {
-                value_to_select -= coin.txout.nValue;
-            }
+            coin.effective_value = coin.txout.nValue - (coin_selection_params.m_subtract_fee_outputs ? 0 : coin_selection_params.m_effective_feerate.GetFee(coin.m_input_bytes));
```

Comment:
Is this correct? I thought it'd be the other way around:
If `m_subtract_fee_outputs` -> subtract `effective feerate` * `m_input_bytes`
else -> subtract 0

```suggestion
            coin.effective_value = coin.txout.nValue - (coin_selection_params.m_subtract_fee_outputs ? coin_selection_params.m_effective_feerate.GetFee(coin.m_input_bytes) : 0);
```